### PR TITLE
Additional sentencepiece args - Byte fallback, split digits, split_on_whitespace

### PR DESCRIPTION
**Pull request by MaximumEntropy**
# What does this PR do ?

Adds some sentencepiece args useful for NLP.

**Collection**: NLP

# Changelog 
- Adds byte_level fallback, split_digits and split_on_whitespace args for spm.

# Usage

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/machine_translation/create_tarred_parallel_dataset.py [hunk 1]
```diff
@@ -86,7 +92,19 @@
...
+        action="store_true",
+        help='Whether to use byte fallback with sentencepiece for BPE tokenization.',
+    )
+    parser.add_argument(
+        '--split_digits', action="store_true", help='Whether to split digits while tokenizing with sentencepiece.'
+    )
+    parser.add_argument(
+        '--no_split_by_whitespace',
+        action="store_true",
+        help='Whether to split digits while tokenizing with sentencepiece.',
```
Comment (aklife97) [inline examples/nlp/machine_translation/create_tarred_parallel_dataset.py:106]: I think the previous helper message got copied here

**Review by aklife97 — APPROVED** [review]
LGTM! 

One concern I have is that with growing changes, our preproc script keeps getting outdated. This means that we can't rely on all preprocessing to be done properly before training if we only provide the preproc directory

**Review by VahidooX — APPROVED** [review]
LGMT!

**Review by aklife97 — APPROVED** [review]
Looks good me too! Cue, Vahid's LGMT

**Pull request merged**